x-pack/filebeat/input/awss3/config.go

```diff
@@ -104,12 +105,45 @@ func (c *config) Validate() error {
 		return errors.New("path_style can only be used when polling non-AWS S3 services")
 	}
 	if c.ProviderOverride != "" && c.NonAWSBucketName == "" {
-		return errors.New("provider can only be overriden when polling non-AWS S3 services")
+		return errors.New("provider can only be overridden when polling non-AWS S3 services")
+	}
+	if c.BackupConfig.NonAWSBackupToBucketName != "" && c.NonAWSBucketName == "" {
+		return errors.New("backup to non-AWS bucket can only be used for non-AWS sources")
+	}
+	if c.BackupConfig.BackupToBucketArn != "" && c.BucketARN == "" {
+		return errors.New("backup to AWS bucket can only be used for AWS sources")
+	}
+	if c.BackupConfig.BackupToBucketArn != "" && c.BackupConfig.NonAWSBackupToBucketName != "" {
+		return errors.New("backup_to_bucket_arn and non_aws_backup_to_bucket_name cannot be used together")
+	}
+	if c.BucketARN != "" || c.NonAWSBucketName != "" {
```

Comment:
let's add another check  when we have `c.QueueURL` and `c.BackupConfig.BackupToBucketArn`
in this case if `if c.BackupConfig.BackupToBucketPrefix == ""` let's log a warning about the fact that if no different prefix it's set up on s3 notification to sqs in AWS this will end in an endless loop